next.config.js

```diff
@@ -26,6 +30,49 @@ module.exports = withCSS(withSourceMaps({
     APP_VERSION: packageJson.version,
     UNLY_SIMPLE_LOGGER_ENV: process.env.APP_STAGE, // Used by @unly/utils-simple-logger - Fix missing staging logs because it believes we're in production
   },
+  experimental: {
+    redirects() {
+      const redirects = [
+        {
+          // Redirect root link with trailing slash to non-trailing slash, avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/',
+          destination: '/:locale',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+        {
+          // Redirect link with trailing slash to non-trailing slash (any depth), avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/:path*/',
+          destination: '/:locale/:path*',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+      ];
+
+      if (process.env.APP_STAGE === 'development') {
+        console.info('Using experimental redirects:', redirects);
+      }
+
+      return redirects;
+    },
+    rewrites() {
+      const rewrites = [
+        {
+          // XXX Doesn't work locally (maybe because of rewrites), but works online
+          source: '/',
+          destination: '/api/autoRedirectToLocalisedPage',
+        },
+        {
+          source: `/:locale((?!${supportedLocales.join('|')})[^/]+)(.*)`,
```

Comment:
Maybe you should block static assets because if an asset is not find inside `public/` folder (404), it is automatically redirected with the locale path. It makes the debugging harder and logs a little wrong. What do you think?

Reply:
I hadn't thought about that. I'll take a look! Thanks for the feedback :)

Reply:
I will notice you if I found a good solution (I was just thinking about a more complex Regexp or maybe a new entry to cancel the redirection) ;)

Reply:
Fix implemented through https://github.com/UnlyEd/next-right-now/pull/42/commits/fad1e2df3f56c969a2838ef830a2b6434838e899

Demo at https://nrn-v1-hyb-mst-aptd-gcms-lcz-sty-c1-cr40srcmj.now.sh/static/images/LOGO_Powered_by_UNLY_monochrome_WHITE.svg2

Reply:
Nice! You need to maintain the list of static but at least it's easier to read and understand than complex regex ;)

Reply:
Yes, it gets converted into a regex eventually, but it's definitely easier to maintain like that.

It could also be automated, but I'm not a fan of overcomplexifying the next.config.js file because it can't run ES6+, and because reading the filesystem on Vercel may be a bit tricky, hardcoded list works just fine.